File: app/src/main/java/com/numero/sojodia/contract/BusScheduleContract.java
package com.numero.sojodia.contract;

import com.numero.sojodia.model.BusTime;
import com.numero.sojodia.presenter.IPresenter;
import com.numero.sojodia.view.IView;

import java.util.List;

public interface BusScheduleContract {

    interface View extends IView<Presenter> {

        void showTkBusTimeList(List<BusTime> busTimeList);

        void showTndBusTimeList(List<BusTime> busTimeList);

        void selectTkCurrentBusPosition(int position);

        void startTkCountDown(BusTime busTime);

        void selectTndCurrentBusPosition(int position);

        void startTndCountDown(BusTime busTime);

        void showTkNextButton();

        void showTkPreviewButton();

        void showTkNoBusLayout();

        void showTndNextButton();

        void showTndPreviewButton();

        void showTndNoBusLayout();

        void hideTkNextButton();

        void hideTkPreviewButton();

        void hideTkNoBusLayout();

        void hideTndNextButton();

        void hideTndPreviewButton();

        void hideTndNoBusLayout();

    }

    interface Presenter extends IPresenter {
        void onTimeChanged(int week);

        void nextTkBus();

        void previewTkBus();

        void nextTndBus();

        void previewTndBus();
    }
}

File: app/src/main/kotlin/com/numero/sojodia/contract/BusScheduleContract.kt
package com.numero.sojodia.contract

import com.numero.sojodia.model.BusTime
import com.numero.sojodia.model.Reciprocate
import com.numero.sojodia.model.Route
import com.numero.sojodia.model.Week
import com.numero.sojodia.presenter.IPresenter
import com.numero.sojodia.view.IView

interface BusScheduleContract {

    interface View : IView<Presenter> {

        fun showTkBusTimeList(busTimeList: MutableList<BusTime>)

        fun showTndBusTimeList(busTimeList: MutableList<BusTime>)

        fun selectTkCurrentBusPosition(position: Int)

        fun startTkCountDown(busTime: BusTime)

        fun selectTndCurrentBusPosition(position: Int)

        fun startTndCountDown(busTime: BusTime)

        fun showTkNextButton()

        fun showTkPreviewButton()

        fun showTkNoBusLayout()

        fun showTndNextButton()

        fun showTndPreviewButton()

        fun showTndNoBusLayout()

        fun hideTkNextButton()

        fun hideTkPreviewButton()

        fun hideTkNoBusLayout()

        fun hideTndNextButton()

        fun hideTndPreviewButton()

        fun hideTndNoBusLayout()

        fun showTimeTableDialog(route: Route, reciprocate: Reciprocate)
    }

    interface Presenter : IPresenter {
        fun onTimeChanged(week: Week)

        fun nextTkBus()

        fun previewTkBus()

        fun nextTndBus()

        fun previewTndBus()

        fun showTimeTableDialog(route: Route)
    }
}
